src/controllers/engine/controllerengine.cpp

```diff
@@ -287,195 +282,96 @@ void ControllerEngine::initializeScripts(const QList<ControllerPreset::ScriptFil
-    // A special version of safeExecute since we're evaluating strings, not actual functions
-    //  (execute() would print an error that it's not a function every time a timer fires.)
-    if (m_pEngine == nullptr) {
-        return false;
-    }
-
-    if (!syntaxIsValid(scriptCode)) {
-        return false;
-    }
-
-    QScriptValue scriptFunction = m_pEngine->evaluate(scriptCode);
-
-    if (checkException()) {
-        qDebug() << "Exception evaluating:" << scriptCode;
-        return false;
-    }
-
-    if (!scriptFunction.isFunction()) {
-        // scriptCode was plain code called in evaluate above
-        return false;
-    }
-
-    return internalExecute(thisObject, scriptFunction, QScriptValueList());
-}
-
-/* -------- ------------------------------------------------------
-Purpose: Evaluate & run script code
-Input:   'this' object if applicable, Code string
-Output:  false if an exception
--------- ------------------------------------------------------ */
-bool ControllerEngine::internalExecute(QScriptValue thisObject, QScriptValue functionObject,
-                                       QScriptValueList args) {
-    if (m_pEngine == nullptr) {
-        qDebug() << "ControllerEngine::execute: No script engine exists!";
+bool ControllerEngine::executeFunction(QJSValue functionObject, QJSValueList args) {
+    VERIFY_OR_DEBUG_ASSERT(!(m_pScriptEngine == nullptr)) {
         return false;
     }
 
     if (functionObject.isError()) {
-        qDebug() << "ControllerEngine::internalExecute:"
+        qDebug() << "ControllerEngine::executeFunction:"
                  << functionObject.toString();
         return false;
     }
 
     // If it's not a function, we're done.
-    if (!functionObject.isFunction()) {
-        qDebug() << "ControllerEngine::internalExecute:"
+    if (!functionObject.isCallable()) {
+        qDebug() << "ControllerEngine::executeFunction:"
                  << functionObject.toVariant()
                  << "Not a function";
         return false;
     }
 
     // If it does happen to be a function, call it.
-    QScriptValue rc = functionObject.call(thisObject, args);
-    if (!rc.isValid()) {
-        qDebug() << "QScriptValue is not a function or ...";
+    QJSValue returnValue = functionObject.call(args);
+
+    if (returnValue.isError()) {
+        showScriptExceptionDialog(returnValue);
         return false;
     }
-
-    return !checkException();
+    return true;
 }
 
-bool ControllerEngine::execute(QScriptValue functionObject,
-                               unsigned char channel,
-                               unsigned char control,
-                               unsigned char value,
-                               unsigned char status,
-                               const QString& group,
-                               mixxx::Duration timestamp) {
-    Q_UNUSED(timestamp);
-    if (m_pEngine == nullptr) {
+bool ControllerEngine::executeFunction(QJSValue functionObject, const QByteArray data) {
```

Comment:
This doesn't really do the same thing as the other `executeFunction`. How about renaming it to `executeFunctionWithData`?

Reply:
Yes it does, it just transforms the QByteArray parameter before passing it to the other executeFunction.